pkg/comp-functions/functions/spksmariadb/script/recreate.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+set -eox pipefail
+
+name="$STS_NAME"
+namespace="$STS_NAMESPACE"
+size="$STS_SIZE"
+release="$RELEASE_NAME"
+
+echo "Checking if the PVC sizes match"
+# Check if delete is necessary
+found=$(kubectl -n "$namespace" get sts "$name" -o json --ignore-not-found)
+
+foundsize=$(echo -En "$found" | jq -r '.spec.volumeClaimTemplates[] | select(.metadata.name=="data") | .spec.resources.requests.storage')
+
+if [[ $foundsize != "$size" ]]; then
+  echo "PVC sizes don't match, deleting sts"
+  # We try to delete the sts and wait for 5s. On APPUiO it can happen that the
+  # deletion with orphan doesn't go through and the sts is stuck with an orphan finalizer.
+  # So if the delete hasn't returned after 5s we forcefully patch away the finalizer.
+  kubectl -n "$namespace" delete sts "$name" --cascade=orphan --ignore-not-found --wait=true --timeout 5s || true
+  kubectl -n "$namespace" patch sts "$name" -p '{"metadata":{"finalizers":null}}' || true
+  # Upause the release so that the sts is recreated. We pause the release to avoid provider-helm updating the release
+  # before the sts is deleted.
+  # Then we first patch the siye to garbage and afterwards to the right size to enforce an upgrade
+  echo "Triggering sts re-creation"
+  kubectl annotate release "$release" "crossplane.io/paused-"
+  kubectl patch release "$release" --type merge -p "{\"spec\":{\"forProvider\":{\"values\":{\"persistence\":{\"size\":\"foo\"}}}}}"
```

Comment:
do we still need that `foo`?

Reply:
Yes. Because provider-helm wouldn't update the release when we just remove the annotation. It needs a change in the values somewhere. This hacky solution looks like the easiest way.